lessons/basics/control-structures.md

```diff
@@ -128,3 +128,49 @@ iex> cond do
 ...> end
 "Catch all"
 ```
+
+## `with`
+
+The special form `with` is useful when you might use a nested `case` statement or situations that cannot cleanly be piped together. We'll start with a simple example of `with` and then look at something more significant:
```

Comment:
Thanks @pragmaticivan!  I'm going to add more to this section based on your suggestion.  I want to refrain from referencing `for` here though since it has not been covered yet in the lessons by this point.